assets/src/components/mapMarkers.tsx

```diff
@@ -0,0 +1,326 @@
+      selectedClass,
+    ]),
+    html: `<svg viewBox="0 0 ${labelBackgroundWidth} ${labelBackgroundHeight}" width="${labelBackgroundWidth}" height="${labelBackgroundHeight}">
+            <rect
+                class="m-vehicle-icon__label-background"
+                width="100%" height="100%"
+                rx="5.5px" ry="5.5px"
+              />
+            <text class="m-vehicle-icon__label" x="50%" y="50%" text-anchor="middle" dominant-baseline="central">
+              ${labelString}
+            </text>
+          </svg>`,
+    iconAnchor: [labelBackgroundWidth / 2, isPrimary ? -16 : -10],
+  })
+}
+
+export const VehicleMarker = ({
+  vehicle,
+  isPrimary,
+  onSelect,
+}: {
+  vehicle: Vehicle
+  isPrimary: boolean
+  onSelect?: (vehicle: Vehicle) => void
+}) => {
+  const [appState] = useContext(StateDispatchContext)
+  const eventHandlers = onSelect ? { click: () => onSelect(vehicle) } : {}
+  const position: LatLngExpression = [vehicle.latitude, vehicle.longitude]
+  const vehicleIcon: Leaflet.DivIcon = makeVehicleIcon(
+    vehicle,
+    isPrimary,
+    appState.userSettings
+  )
+  const labelIcon: Leaflet.DivIcon = makeLabelIcon(
+    vehicle,
+    isPrimary,
+    appState.userSettings,
+    appState.selectedVehicleOrGhost?.id || ""
+  )
+  const zIndexOffset = isPrimary ? 2000 : 0
+  return (
+    <>
+      <Marker
+        position={position}
+        icon={vehicleIcon}
+        eventHandlers={eventHandlers}
+        zIndexOffset={zIndexOffset}
+      />
+      <Marker
+        position={position}
+        icon={labelIcon}
+        eventHandlers={eventHandlers}
+        zIndexOffset={zIndexOffset}
+      />
+    </>
+  )
+}
+
+const makeTrainVehicleIcon = ({ bearing }: TrainVehicle): Leaflet.DivIcon => {
+  const centerX = 24
+  const centerY = 24
+  return Leaflet.divIcon({
+    html: `<svg xmlns="http://www.w3.org/2000/svg" role="img" viewBox="0 0 36 36">
+        <g transform="rotate(${bearing}, ${centerX}, ${centerY})">
+          <path fill="#fff" d="m42.88 45.83a2.1 2.1 0 0 1 -.87-.19l-15.92-7.17a5.23 5.23 0 0 0 -2.09-.47 5.14 5.14 0 0 0 -2.08.44l-15.92 7.2a2.1 2.1 0 0 1 -.87.19 2.14 2.14 0 0 1 -1.76-1 2 2 0 0 1 -.12-2l18.86-40.83a2.08 2.08 0 0 1 3.78 0l18.87 40.87a2 2 0 0 1 -.12 2 2.14 2.14 0 0 1 -1.76.96z"/>
+        </g>
+    </svg>`,
+    className: "m-vehicle-map__train-icon",
+  })
+}
+
+export const TrainVehicleMarker = ({
+  trainVehicle,
+}: {
+  trainVehicle: TrainVehicle
+}) => {
+  const position: LatLngExpression = [
+    trainVehicle.latitude,
+    trainVehicle.longitude,
+  ]
+  const icon: Leaflet.DivIcon = makeTrainVehicleIcon(trainVehicle)
+  return <Marker position={position} icon={icon} />
+}
+
+export const strokeOptions = ({ color }: Shape): object =>
+  color
+    ? {
+        color,
+        opacity: 1.0,
+        weight: 4,
+      }
+    : {
+        color: "#4db6ac",
+        opacity: 0.6,
+        weight: 6,
+      }
+
+const StopMarker = ({ stop }: { stop: Stop }) => (
```

Comment:
**nitpick:** it probably makes more sense for this to go after `StopOrStationMarker` but before `StationMarker`.